packages/venia-concept/src/ServiceWorker/Utilities/htmlHandler.js

```diff
@@ -142,17 +143,75 @@ const cloneRequestWithDiffURL = (request, url) =>
           })
         : request;
 
+/**
+ * Function validates if the script files mentioned in the cached
+ * HTML file are cached. If not, it returns false.
+ *
+ * @returns Promise<Boolean>
+ */
+const areAllHTMLLinksCached = async () => {
+    const cachedResponseObj = await caches.match('/');
+
+    if (cachedResponseObj) {
+        const cachedResponse = await cachedResponseObj.text();
+        const parsedCacheResponse = parse(cachedResponse, {
+            style: true
+        });
+        const scriptFiles = parsedCacheResponse
+            .querySelectorAll('script')
+            .map(generateScriptResource)
+            .filter(identity);
+
+        const cachedScriptFileResponses = await Promise.all(
+            scriptFiles.map(file => caches.match(file))
+        );
+
+        return cachedScriptFileResponses.every(file => !!file);
+    } else {
+        return Promise.resolve(false);
```

Comment:
`async` automatically wraps your returns in a Promise, this shouldn't be needed.

```suggestion
        return false;
```

Reply:
Good to know. I wasn't aware.